rpcclient/infrastructure.go

```diff
@@ -756,14 +759,31 @@ func (c *Client) handleSendPostMessage(jReq *jsonRequest) {
 	if !c.config.DisableTLS {
 		protocol = "https"
 	}
-	url := protocol + "://" + c.config.Host
 
 	var (
 		err, lastErr error
 		backoff      time.Duration
 		httpResponse *http.Response
 	)
 
+	parsedAddr, err := ParseAddressString(c.config.Host, "")
```

Comment:
Since we're not using the default port (it's always empty), we should probably remove that argument and update the two new functions accordingly.